text/0074/0074-entity-slicing.md

```diff
@@ -0,0 +1,566 @@
+         if <expr> { <expr> } { <expr> }
+         ... all other Cedar operators not mentioned by datapath-expr
+         
+<datapath-expr> = <datapath-expr>.<field>
+                  <datapath-expr> has <field>
+                  <variable>
+                  <entity literal>
+                  
+<variable> = principal 
+             resource
+             action
+             context
+                  
+```
+
+
+The grammar partitions Cedar expressions into **datapath expressions** and all other Cedar operators.
+On the LHS of `in`, `.`, and `has`, expressions involving `in`, `+`, `if-then-else`, and most other Cedar functions/operators are not allowed; only Cedar variables, entity literals, and `.` or `has` expressions are allowed.
+
+The partition is reasonable since, in practice, users do not interleave datapath expressions with other operators.
+Here are examples of Cedar expressions rejected by this grammar:
+
+
+```
+{ "myfield": principal.name }.myfield
+(if principal.ismanager then { "myfield" : 2 } else { "myfield": 3 }).myfield
+(if principal has mobilePhone then principal.mobilePhone else principal.workPhone).zipCode
+```
+
+
+Now that we have this partition, computing all the access paths is fairly simple:
+
+1. First, find all datapath expressions. These can be translated into access paths directly.
+2. Second, handle all instances of equality between records. For equality between records (`==`) or set containment where the elements are records (`.contains`, `.containsAny`, and `.containsAll`)
+        1. Find all access paths that are children of these operators
+            1. Following the schema, fully load all fields for the leaf of the access path
+3. Finally, handle instances where the ancestors in the entity hierarchy are required. Annotate the left-hand-side of the `in` operator with the `ancestors_required` flag.
+
+## 
+
+## SimplifiedEntityLoader API
+
+Having the entity manifest enables different data loading strategies that are less precise, but easy to implement.
+The easiest is the `SimpleEntityLoader`  api shown here. Users need only write a function for loading entities via their IDs, and Cedar handles determining which entities need to be loaded.
+
+
+```
+/// Implement this trait to load entities based on
+/// the entity manifest.
+/// This entity loader is called "Simple" for two reasons:
+/// 1) First, it may cause database consistency challenges- `load_entity` is called multiple times.
+/// 2) Second, it is not precise- the entity manifest only requires some
+/// fields to be loaded, but this loader always loads all fields.
+pub trait SimpleEntityLoader {
+    /// Simple entity loaders must implement `load_entity`,
+    /// a function that loads entities based on their `EntityUID`s.
+    /// For each element of `entity_ids`, returns the corresponding
+    /// [`Entity`] in the output vector.
+    fn load_entity(&mut self, entity_ids: impl IntoIterator<Item = &EntityUID>) -> impl Iterator<Item = Entity>;
```

Comment:
nit: Kind of a weird name given that it accepts multiple entity IDs.

Reply:
fixed